analytics-core/src/main/java/com/segment/analytics/AnalyticsContext.java

```diff
@@ -115,20 +116,21 @@
    * {@link Context}. The {@link Analytics} client can be called from anywhere, so the returned
    * instances is thread safe.
    */
-  static synchronized AnalyticsContext create(Context context, Traits traits) {
+  static synchronized AnalyticsContext create(Context context, Traits traits,
+                                              boolean collectDeviceId) {
     AnalyticsContext analyticsContext =
         new AnalyticsContext(new NullableConcurrentHashMap<String, Object>());
     analyticsContext.putApp(context);
-    analyticsContext.putDevice(context);
+    analyticsContext.setTraits(traits);
+    analyticsContext.putDevice(context, collectDeviceId);
     analyticsContext.putLibrary();
     analyticsContext.put(LOCALE_KEY,
-        Locale.getDefault().getLanguage() + "-" + Locale.getDefault().getCountry());
+            Locale.getDefault().getLanguage() + "-" + Locale.getDefault().getCountry());
```

Comment:
The CI is failing with a checkstyle error — likely because of this line. You can run `./gradlew build` locally to run checkstyle and verify the cause.